### PR TITLE
Enhancement: Add RW assignee as default sort option

### PR DESCRIPTION
**Pull request by jhoward1994**
### What does it do?

Replaces RW stage sort with assignee as that is what is relevant to this branch

I thought it was best to replace the stage stuff rather than add to it so we keep the branches separate during development. wdyt?

This PR is FE only

### Why is it needed?

Adding default sort ability on RW assignee

### CODE REVIEW

**Comment by jhoward1994** [comment]
@Marc-Roig also pinging you here because I've made the changes to use `getDisplayName` as you mentioned https://github.com/strapi/strapi/pull/17476#discussion_r1276444194 in this PR

File: packages/core/admin/ee/admin/content-manager/pages/ListSettingsView/constants.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
-export const REVIEW_WORKFLOW_STAGE_SORT_OPTION_NAME = {
-  value: 'strapi_stage[name]',
+export const REVIEW_WORKFLOW_ASSIGNEE_SORT_OPTION_NAME = {
```
Comment (gu-stav) [inline packages/core/admin/ee/admin/content-manager/pages/ListSettingsView/constants.js:1]: Am I missing something - why would you want to overwrite the stage sorting? Don't you want to add a new option?

File: packages/core/admin/ee/admin/content-manager/pages/ListView/ReviewWorkflowsColumn/ReviewWorkflowsAssigneeEE.js [hunk 2]
```diff
@@ -4,48 +4,22 @@ import { Typography } from '@strapi/design-system';
...
-  displayname: '',
+  user: { firstname: '', lastname: '', username: '' },
 };
 
 ReviewWorkflowsAssigneeEE.propTypes = {
-  firstname: PropTypes.string,
-  lastname: PropTypes.string,
-  displayname: PropTypes.string,
+  user: PropTypes.shape({
+    firstname: PropTypes.string,
```
Comment (gu-stav) [inline packages/core/admin/ee/admin/content-manager/pages/ListView/ReviewWorkflowsColumn/ReviewWorkflowsAssigneeEE.js:21]: Firstname is a required prop, isn't it?

File: packages/core/admin/ee/admin/content-manager/pages/ListSettingsView/constants.js (hunk 1 above)
Comment (gu-stav) [reply packages/core/admin/ee/admin/content-manager/pages/ListSettingsView/constants.js]: Fixed & rebeased

**Comment by gu-stav** [comment]
@Marc-Roig Do you feel comfortable to review this (to finish the sprint)? :)

**Review by gu-stav — APPROVED** [review]

**Review by Marc-Roig — APPROVED** [review]

**Pull request merged**